app/models/container_node.rb

```diff
@@ -8,6 +8,7 @@ class ContainerNode < ActiveRecord::Base
   has_many   :container_conditions, :class_name => ContainerCondition, :as => :container_entity, :dependent => :destroy
   has_many   :containers, :through => :container_groups
   has_many   :container_services, -> { distinct }, :through => :container_groups
+  has_many   :container_routes, :through => :container_services
```

Comment:
@moolitayer add a test here as well (demonstrating that there's no need for `distinct`).